source/scripts/notesEditor.js

```diff
@@ -69,13 +63,21 @@ function addEditToggle(preview, id){
         editButton.innerHTML = 'Edit';
     } else {
         editButton.innerHTML = 'View';
+        addNewNoteButtons(parseInt(id), db);
     }
     buttonSection.appendChild(editButton);
-    editButton.addEventListener('click', () => {
+    editButton.addEventListener('click', async () => {
+        preview = !preview;
+        let note = await getNoteFromStorage(db, parseInt(id));
```

Comment:
We should just toggle the text areas from editable to not editable and vice versa within this method so that we don't have to make an extra call to the database and call addNotesToDocument() again.

Reply:
Otherwise it seems like it would be almost just as efficient to navigate to a new url